qiskit/algorithms/optimizers/adam_amsgrad.py

```diff
@@ -114,6 +114,13 @@ def __init__(
                 writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
                 writer.writeheader()
 
+    @property
+    def settings(self) -> Dict[str, Any]:
+        settings = {"name": "ADAM"}
```

Comment:
Is there are reason we do not just use the classname rather than coding a string directly - in the few I have seen this looks identical

Reply:
see resolved comment above 🙂